File: Fidget Spinner.java
import java.util.*;

public class Main {
    static final int MOD = 1000000007;

    public static int ways(int sum) {
        // Write your logic here.
        if(sum == 0)return 1;
        if(sum < 0) return 0;
        int way = 0;

        for(int i=1;i<=6;i++){
            way = way+ways(sum-i)%MOD;
        }
        return way%MOD; // Placeholder
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int result = ways(n);
        System.out.println(result);
    }
}

File: Fidget Spinner.cpp
#include <iostream>
#include <vector>
#define MOD 1000000007

using namespace std;

int ways(int N) {
    vector<int> dp(N + 1, 0);
    dp[0] = 1; // Base case

    for (int i = 1; i <= N; i++) {
        for (int j = 1; j <= 6; j++) {
            if (i - j >= 0) {
                dp[i] = (dp[i] + dp[i - j]) % MOD;
            }
        }
    }
    return dp[N];
}

int main() {
    int n;
    cin >> n;
    cout << ways(n) << endl;
    return 0;
}
